ui/src/shared/components/ColorScaleDropdown.js

```diff
@@ -0,0 +1,117 @@
+import FancyScrollbar from 'shared/components/FancyScrollbar'
+
+import {LINE_COLOR_SCALES} from 'src/shared/constants/graphColorPalettes'
+
+class ColorScaleDropdown extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      expanded: false,
+    }
+  }
+
+  handleToggleMenu = () => {
+    const {disabled} = this.props
+
+    if (disabled) {
+      return
+    }
+    this.setState({expanded: !this.state.expanded})
+  }
+
+  handleClickOutside = () => {
+    this.setState({expanded: false})
+  }
+
+  handleDropdownClick = colorScale => () => {
+    this.props.onChoose(colorScale)
+    this.setState({expanded: false})
+  }
+
+  generateGradientStyle = colors => ({
+    background: `linear-gradient(to right, ${colors[0].hex} 0%,${
+      colors[1].hex
+    } 50%,${colors[2].hex} 100%)`,
+  })
+
+  render() {
+    const {expanded} = this.state
+    const {selected, disabled, stretchToFit} = this.props
+
+    const dropdownClassNames = classnames('color-dropdown', {
+      open: expanded,
+      'color-dropdown--stretch': stretchToFit,
+    })
+    const toggleClassNames = classnames(
+      'btn btn-sm btn-default color-dropdown--toggle',
+      {active: expanded, 'color-dropdown__disabled': disabled}
+    )
+
+    return (
+      <div className={dropdownClassNames}>
+        <div
+          className={toggleClassNames}
+          onClick={this.handleToggleMenu}
+          disabled={disabled}
+        >
+          <div
+            className="color-dropdown--swatches"
+            style={this.generateGradientStyle(selected)}
+          />
+          <div className="color-dropdown--name">{selected[0].name}</div>
+          <span className="caret" />
+        </div>
+        {expanded ? (
+          <div className="color-dropdown--menu">
+            <FancyScrollbar autoHide={false} autoHeight={true}>
+              {LINE_COLOR_SCALES.map(colorScale => (
+                <div
+                  className={
+                    colorScale.name === selected[0].name
+                      ? 'color-dropdown--item active'
+                      : 'color-dropdown--item'
+                  }
+                  key={uuid.v4()}
+                  onClick={this.handleDropdownClick(colorScale)}
+                >
+                  <div
+                    className="color-dropdown--swatches"
+                    style={this.generateGradientStyle(colorScale.colors)}
+                  />
+                  <span className="color-dropdown--name">
+                    {colorScale.name}
+                  </span>
+                </div>
+              ))}
+            </FancyScrollbar>
+          </div>
+        ) : null}
+      </div>
+    )
+  }
+}
+
+const {arrayOf, bool, func, shape, string} = PropTypes
+
+ColorScaleDropdown.propTypes = {
+  selected: arrayOf(
+    shape({
+      type: string.isRequired,
+      hex: string.isRequired,
+      id: string.isRequired,
+      name: string.isRequired,
+    }).isRequired
```

Comment:
If this shape is the same as all the rest of the color types, you should move a general schema definition into `shared/schemas.js` for now, ahead of moving these files to typescript. Then refer to that schema in all places you need a color